internal/framework/resources/registered_domain/resource_test.go

```diff
@@ -0,0 +1,226 @@
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccRegisteredDomainResource_WithExtendedAttrs(t *testing.T) {
+	// var resourceID string
+	// Get convert the contact id to int
+	contactID, err := strconv.Atoi(os.Getenv("DNSIMPLE_CONTACT_ID"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	domainName := utils.RandomString(62) + ".eu"
+	resourceName := "dnsimple_registered_domain.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheckRegisteredDomain(t) },
+		ProtoV6ProviderFactories: test_utils.TestAccProtoV6ProviderFactories(),
+		CheckDestroy:             testAccCheckRegisteredDomainResourceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRegisteredDomainResourceConfig_WithExtendedAttrs(domainName, contactID),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", domainName),
+					resource.TestCheckResourceAttr(resourceName, "state", "registered"),
+					resource.TestCheckResourceAttrSet(resourceName, "domain_registration.id"),
+					resource.TestCheckResourceAttr(resourceName, "auto_renew_enabled", "false"),
+					resource.TestCheckResourceAttr(resourceName, "whois_privacy_enabled", "false"),
+					resource.TestCheckResourceAttr(resourceName, "dnssec_enabled", "false"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportStateIdFunc: testAccRegisteredDomainImportStateIDFunc(resourceName),
+				ImportState:       true,
+				ImportStateVerify: false,
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccRegisteredDomainResource_WithOptions(t *testing.T) {
+	// Get convert the contact id to int
+	contactID, err := strconv.Atoi(os.Getenv("DNSIMPLE_CONTACT_ID"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	domainName := utils.RandomString(62) + ".com"
+	resourceName := "dnsimple_registered_domain.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheckRegisteredDomain(t) },
+		ProtoV6ProviderFactories: test_utils.TestAccProtoV6ProviderFactories(),
+		CheckDestroy:             testAccCheckRegisteredDomainResourceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRegisteredDomainResourceConfig_WithOptions(domainName, contactID, false, true, true),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", domainName),
+					resource.TestCheckResourceAttr(resourceName, "state", "registered"),
+					resource.TestCheckResourceAttrSet(resourceName, "domain_registration.id"),
+					resource.TestCheckResourceAttr(resourceName, "auto_renew_enabled", "false"),
+					resource.TestCheckResourceAttr(resourceName, "whois_privacy_enabled", "true"),
+					resource.TestCheckResourceAttr(resourceName, "dnssec_enabled", "true"),
+				),
```

Comment:
Same as above.

Reply:
This test case has no extended attributes, all attributes are accounted for in the spec.